docs/src/description/organization.md

```diff
@@ -0,0 +1,77 @@
+# Code organization
+
+AirScript project can consist of one or more modules, each module located in a separate file with a `.air` extension. For projects consisting of multiple modules, one module must be declared as the root module, and all other modules must be declared as supporting modules.
+
+Currently, all modules must be located in a single directory, but in the future this limitation will be removed.
+
+All modules must start with a module name declaration followed by a set of source sections. These sections describe both the metadata and constraint evaluation logic for the AIR. Depending on the module type, some source sections may be required, be optional, a may not be allowed. The table below summarizes this information.
+
+| Section                                                                               | Root module | Supporting module |
+| ------------------------------------------------------------------------------------- | :---------: | :---------------: |
+| [constants](./declarations.md#constants-const)                                        | optional    | optional          |
+| [tace columns](./declarations.md#execution-trace-trace_columns)                       | required    | not allowed       |
+| [public inputs](./declarations.md#public-inputs-public_inputs)                        | required    | not allowed       |
+| [periodic columns](./declarations.md#periodic-columns-periodic_columns)               | optional    | optional          |
+| [random values](./declarations.md#random-values-random_values)                        | optional    | not allowed       |
+| [boundary constraints](./constraints.md#boundary-constraints-boundary_constraints)    | required    | not allowed       |
+| [integrity constraints](./constraints.md#integrity-constraints-integrity_constraints) | required    | not allowed       |
+| [evaluators](./evaluators.md)                                                         | optional    | required          |
+
+Note that constants and evaluators are not really distinct sections but rather a set of declarations which can be done in-between any other sections.
+
+### Root module
+A root module defines an entrypoint into an AirScript project. It must start with a name declaration which consists of a `def` keyword followed by the name of the AIR project. For example:
+```
+def ExampleAir
```

Comment:
NOTE: Module names must match the file name they are defined in, though without fickle checking the code, I do think I made the search case-insensitive. Just something to point out anyway. I'd also recommend using the `snake_case` naming convention for that reason.

Reply:
This is true only for library modules, right? I was able to compile a root module named `ExampleAir` from `example.air` file without issues.

Reply:
That's a good point, it applies to import resolution/loading, so is really only relevant for library modules. That said, having module names that don't match the file name (or some other convention) makes it difficult to find modules programmatically, which might be a nice property to preserve. Definitely not essential as long as the typical workflow is compiling by passing a file path to the root module (or a string containing a root module definition).